### PR TITLE
[Plugin] Split object identifier and add more ride object fields

### PR DESCRIPTION
**Pull request by IntelOrca**
Discussed here:
https://github.com/OpenRCT2/OpenRCT2/pull/11489#issuecomment-620014113

And most of the ride fields. @spacek531

### CODE REVIEW

File: src/openrct2/object/ObjectFactory.cpp [hunk 1]
```diff
@@ -104,15 +104,20 @@ class ReadObjectContext : public IReadObjectContext
     }
 
     ReadObjectContext(
-        IObjectRepository& objectRepository, const std::string& objectName, bool loadImages,
+        IObjectRepository& objectRepository, const std::string& identifier, bool loadImages,
         const IFileDataRetriever* fileDataRetriever)
         : _objectRepository(objectRepository)
         , _fileDataRetriever(fileDataRetriever)
-        , _objectName(objectName)
+        , _identifier(identifier)
```
Comment (Gymnasiast) [inline src/openrct2/object/ObjectFactory.cpp:111]: Is this the new identifier, or the legacy one?

File: src/openrct2/localisation/LanguagePack.cpp [hunk 2]
```diff
@@ -198,15 +198,14 @@ class LanguagePack final : public ILanguagePack
         }
     }
 
-    rct_string_id GetObjectOverrideStringId(const char* objectIdentifier, uint8_t index) override
+    rct_string_id GetObjectOverrideStringId(const std::string_view& objectIdentifier, uint8_t index) override
```
Comment (Gymnasiast) [inline src/openrct2/localisation/LanguagePack.cpp:201]: This takes a legacy identifier, right? If so: perhaps best to change the parameter name.

File: src/openrct2/object/ObjectFactory.cpp (hunk 1 above)
Comment (IntelOrca) [reply src/openrct2/object/ObjectFactory.cpp:111]: I think it is new for JSON, old for DAT, but we only use it for logging at the moment.

File: src/openrct2/localisation/LanguagePack.cpp (hunk 2 above)
Comment (IntelOrca) [reply src/openrct2/localisation/LanguagePack.cpp:201]: sure

**Pull request merged**